staticjinja/dep_graph.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding:utf-8 -*-
+
+"""
+Dependency graph for staticjina
+"""
+
+from copy import deepcopy
+
+
+class DepGraph(object):
+    """
+    A directed graph which will handle dependencies between templates and data
+    files in a site.
+
+    :param site:
+        A :class:`Site <Site>` object.
+
+    """
+    def __init__(self, site):
+        self.parents = dict(
+                (s, set()) for s in site.jinja_names + site.data_names
+                )
+        self.children = deepcopy(self.parents)
+
+        for filename in site.jinja_names:
+            self.parents[filename] = site.get_file_dep(filename)
+            for d in self.parents[filename]:
+                self.children[d].add(filename)
```

Comment:
I don't know. To me the coexistence of Site.__init__ and make_site means a lot of code duplication. I was a bit puzzled by this when working on the improved context PR, I could only see all my copy and paste and no benefit. But of course I have very few experience in serious python so I'm totally ready to believe you.

If you want a more generic graph class then you can pass the parents dict and the get_file_dep method to the constructor? You could even pass the parents *and* children dicts but it would be rather close to getting rid of DepGraph and put everything in Site and Reloader (which is what I did initially). In update you could then pass new_parents.